LEAF_Request_Portal/scripts/process_queries.php

```diff
@@ -5,13 +5,28 @@
 
 $currDir = dirname(__FILE__);
 
-require_once '../globals.php';
+require_once $currDir.'/../globals.php';
 require_once LIB_PATH . '/loaders/Leaf_autoloader.php';
 
 $login->setBaseDir('../');
 $login->loginUser();
 
-
 $form = new Portal\Form($db, $login);
 
+// last process time
+$vars = [
+    //':lastProcess' => strtotime("30 minutes ago")
+    ':lastProcess' => time()
+];
+
+$processQueryTotalSQL = 'SELECT id,userID,`url` FROM process_query WHERE lastProcess <= :lastProcess group by `url`';
+$processQueryTotalRes = $db->prepared_query($processQueryTotalSQL, $vars);
 

```

Comment:
Just thinking of the future, if we want to get to a more MVC style application then we should start moving queries from flat files like this one into classes. I personally like class names that match Db table names but that is for the team to decide how we want to do that.

Not sure if this query would ever get used again somewhere else but regardless having it here ensures that it can never be used again for anything else. In a class it could be used again if needed without having to write it.

Technically, the if and foreach that follows could be placed in a class as well. Placing this inside a method in a class makes it easier to test. It could have a unit test built to do that. That would leave the flat file really to just kick this all off and let the heavy lifting be done by the class.

Just some thoughts for us to be thinking about.

Reply:
I agree, but as discussed not sure if we want to do things as model per tableor do model perfunctionality. They have their pros and cons. I prefer the model per table since it takes away ambiguity.